src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -0,0 +1,220 @@
+            $doctrineEnvelope = $this->executeQuery(
+                $query->getSQL().' '.$this->driverConnection->getDatabasePlatform()->getWriteLockSQL(),
+                [
+                    ':now' => (\DateTime::createFromFormat('U.u', microtime(true)))->format('Y-m-d H:i:s.u'),
+                    ':queue' => $this->configuration['queue'],
+                    ':redeliver_limit' => $redeliverLimit->format('Y-m-d H:i:s'),
+                ]
+            )->fetch();
+
+            if (false === $doctrineEnvelope) {
+                $this->driverConnection->commit();
+
+                return null;
+            }
+
+            $type = Type::getType(Type::JSON);
+            $doctrineEnvelope['headers'] = $type->convertToPHPValue($doctrineEnvelope['headers'], $this->driverConnection->getDatabasePlatform());
+
+            $sql = $this->driverConnection->createQueryBuilder()
+                ->update($this->configuration['table_name'])
+                ->set('delivered_at', ':delivered_at')
+                ->where('id = :id');
+            $this->executeQuery($sql, [
+                ':id' => $doctrineEnvelope['id'],
+                ':delivered_at' => (\DateTime::createFromFormat('U.u', microtime(true)))->format('Y-m-d H:i:s.u'),
+            ]);
+
+            $this->driverConnection->commit();
+
+            return $doctrineEnvelope;
+        } catch (\Throwable $e) {
+            $this->driverConnection->rollBack();
+
+            throw $e;
+        }
+    }
+
+    public function ack($id): bool
+    {
+        return $this->driverConnection->delete($this->configuration['table_name'], ['id' => $id]) > 0;
+    }
+
+    public function reject($id): bool
+    {
+        return $this->driverConnection->delete($this->configuration['table_name'], ['id' => $id]) > 0;
+    }
+
+    private function executeQuery(string $sql, array $parameters = [])
+    {
+        $stmt = null;
+        try {
+            $stmt = $this->driverConnection->prepare($sql);
+            $stmt->execute($parameters);
+        } catch (TableNotFoundException $e) {
+            // create table
+            if (!$this->driverConnection->isTransactionActive() && $this->configuration['auto_setup']) {
+                $this->setup();
+            }
+            // statement not prepared ? SQLite throw on exception on prepare if the table does not exist
+            if (null === $stmt) {
+                $stmt = $this->driverConnection->prepare($sql);
+            }
+            $stmt->execute($parameters);
+        }
+
+        return $stmt;
+    }
+
+    private function setup(): void
+    {
+        $synchronizer = new SingleDatabaseSynchronizer($this->driverConnection);
+        $synchronizer->updateSchema($this->getSchema(), true);
+    }
+
+    private function getSchema(): Schema
+    {
+        $schema = new Schema();
+        $table = $schema->createTable($this->configuration['table_name']);
+        $table->addColumn('id', Type::BIGINT)
```

Comment:
Can we use a UUID instead?

Reply:
I was wondering about that too. Are there any real differences/pros/cons? Looking at 2 other libs, one uses UUID's and one uses id's.

Reply:
@sroze The only problem is PostgreSQL. It requires an extension to generate uuids: `uuid-ossp`.

Maybe we can have an option `id_strategy` to configure if we want to use AUTO_INCREMENT or uuids. Default would be AUTO_INCREMENT.

Then we just need to document that for using uuid on Postgres databases the extension is required.

Reply:
I see. I like the `id_strategy` you've built, it's reasonable. Though, the fact that it's not available on all drivers was a good excuse for just not building it :)

Reply:
Ok so I can remove the uuid support ? Or maybe we can use `ramsey/uuid` as explained here : https://github.com/symfony/symfony/pull/29007#issuecomment-478245942 . :thinking:

Reply:
Let's make it simple and not have UUID for now then, it will simplify the PR and can be added later if such a thing is needed.